web/src/utilities/utilities.js

```diff
@@ -611,6 +611,38 @@ export const Utilities = {
     }
   },
 
+  isInitialAppInstall(app) {
+    if (!app) {
+      return false;
+    }
+
+    if (app.downstream?.currentVersion) {
+      // a version has already been deployed
+      return false;
+    }
+
+    if (
+      !app.downstream?.pendingVersions ||
+      app.downstream.pendingVersions.length === 0
+    ) {
+      // there's no version that's been deployed or is pending, so it's an initial install
+      return true;
+    }
+
+    const firstPendingVersion = app.downstream.pendingVersions[0];
```

Comment:
this might be an edge case, but what if there's more than 1 pending version?

Reply:
i did remember a bug where there was once two pending versions created during initial install. it seems very rare, but do you think we should be taking the last element in the list instead of the first? should we be iterating and checking all of them?

Reply:
honestly, it's probably fine regardless. this doesn't affect any critical functionality. i'd be okay with leaving it like this for now.

Reply:
or, simply return `false` if there's more than 1 pending version?

Reply:
i'll go with that second one for now. the end-user might probably be stuck anyway if there's >1 pending version on initial install for some reason.